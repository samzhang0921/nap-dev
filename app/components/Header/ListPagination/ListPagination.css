.pagenationBox {
  order:3;
  text-align: right;
}
.pagenationBoxUl {
  list-style: none;
  display: flex;
  margin: 0px;
}
@media (max-width: 949px) {
  .pagenationBox {
    order: 3;
  }
}
