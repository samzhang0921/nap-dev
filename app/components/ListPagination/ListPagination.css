.ListPagination {
  flex: 1;
  text-align: right;
  align-self: baseline;
}
