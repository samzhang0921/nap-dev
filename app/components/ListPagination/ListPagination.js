import React, { Component } from 'react';
import style from './ListPagination.css';
const ListPagination = props => {
  return (
    <div className={style.ListPagination}>
      ListPagination
    </div>
  );
};

export default ListPagination;
