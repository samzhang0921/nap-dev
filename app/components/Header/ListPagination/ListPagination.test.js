import { mount } from 'enzyme';
import ListPagination from './ListPagination';
import React from 'react';
import sinon from 'sinon';

describe('ListPagination', () => {
  const totalPage = 10;
  const currentPage = 1;
  const updateOffset = sinon.spy();

  const mountWrapper = mount((
      <ListPagination totalPage={totalPage} currentPage={currentPage} updateOffset={updateOffset} />
  ));

  it('should render a ul', () => {
      expect(mountWrapper.find('ul')).toHaveLength(1);
  });

  it('should have correct props', () => {
      expect(mountWrapper.props().totalPage).toEqual(10);
  });

  it('should trigger updateOffset when its clicked', () => {
      mountWrapper.find('li').first().simulate('click');
      expect(updateOffset).toHaveProperty('callCount', 1);
  });
});
