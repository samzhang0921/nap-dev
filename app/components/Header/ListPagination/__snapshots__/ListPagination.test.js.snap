// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPagination should match its snapshot with props 1`] = `
<div
  className="pagenationBox"
>
  <ul
    className="pagenationBoxUl"
  >
    <li
      className="pagination--1 pagenationBoxLi"
      onClick={[Function]}
    >
      <span>
        
      </span>
    </li>
    <li
      className="pagenationBoxCurrentPage"
      onClick={[Function]}
    >
      <span>
        1
      </span>
    </li>
    <li
      className="pagination-2 pagenationBoxLi"
      onClick={[Function]}
    >
      <span>
        2
      </span>
    </li>
    <li
      className="pagination-3 pagenationBoxLi"
      onClick={[Function]}
    >
      <span>
        3
      </span>
    </li>
    <li
      className="pagination-dot pagenationBoxLi"
      onClick={[Function]}
    >
      <span>
        ...
      </span>
    </li>
    <li
      className="pagination-10 pagenationBoxLi"
      onClick={[Function]}
    >
      <span>
        10
      </span>
    </li>
    <li
      className="pagination-0 pagenationBoxLi"
      onClick={[Function]}
    >
      <span>
        
      </span>
    </li>
  </ul>
</div>
`;

exports[`ListPagination should mount without props  1`] = `
<ListPagination>
  <div
    className="pagenationBox"
  >
    <ul
      className="pagenationBoxUl"
    >
      <PageNumber
        page={-1}
      >
        <li
          className="pagination--1 pagenationBoxLi"
          onClick={[Function]}
        >
          <span />
        </li>
      </PageNumber>
      <PageNumber
        page={1}
      >
        <li
          className="pagination-1 pagenationBoxLi"
          onClick={[Function]}
        >
          <span>
            1
          </span>
        </li>
      </PageNumber>
      <PageNumber
        page="..."
      >
        <li
          className="pagination-dot pagenationBoxLi"
          onClick={[Function]}
        >
          <span>
            ...
          </span>
        </li>
      </PageNumber>
      <PageNumber
        page={0}
      >
        <li
          className="pagination-0 pagenationBoxLi"
          onClick={[Function]}
        >
          <span />
        </li>
      </PageNumber>
    </ul>
  </div>
</ListPagination>
`;
